pkg/service/operation/storage/bolt.go

```diff
@@ -43,18 +44,31 @@ func (b BoltOperationStorage) GetOperation(id string) (*StoredOperation, error)
 	return &stored, nil
 }
 
-func (b BoltOperationStorage) GetOperations() ([]StoredOperation, error) {
+func (b BoltOperationStorage) GetOperations(filter filtering.Filter) ([]StoredOperation, error) {
 	operations, err := b.db.ReadAll(namespace)
 	if err != nil {
 		return nil, util.LoggingErrorMsgf(err, "could not get all operations")
 	}
+
+	shouldInclude, err := storage.NewIncludeFunc(filter)
+	if err != nil {
+		return nil, err
+	}
 	stored := make([]StoredOperation, 0, len(operations))
 	for i, manifestBytes := range operations {
 		var nextOp StoredOperation
 		if err = json.Unmarshal(manifestBytes, &nextOp); err != nil {
 			logrus.WithError(err).WithField("idx", i).Warnf("Skipping operation")
 		}
-		stored = append(stored, nextOp)
+		include, err := shouldInclude(nextOp)
+		// We explicitly ignore evaluation errors and simply include them in the result.
+		if err != nil {
+			stored = append(stored, nextOp)
+			continue
+		}
+		if include {
+			stored = append(stored, nextOp)
+		}
```

Comment:
could this be changed to something like this?

```suggestion
		if err != nil {
			logrus.WithErr(err).Error(...)
		}
		stored = append(stored, nextOp)
```

Reply:
Errors are already logged inside the `NewIncludeFunc`

Separately, your suggestion doesn't check the `include` variable, which would break the filtering behavior.

Reply:
how does it break the behavior? as I read it your code does the same behavior whether or not the include var is checked

Reply:
Suppose that the `shouldInclude` function returns:
 * `include` equal to `false`
 * `err` equal to `nil` 

Under the above scenario, your suggestion would include `nextOp`; the current implementation would not.

Reply:
ah, gotcha. guess it's fine as is but the continue isn't too useful and you could combine the cases